mvc/src/main/java/nextstep/mvc/handlerMapping/AnnotationHandlerMapping.java

```diff
@@ -50,22 +52,20 @@ public Object getHandler(final HttpServletRequest request) {
         return handlerExecutions.get(handlerKey);
     }
 
-    private void setHandlerExecutions(final Class<?> handler, final Set<Method> annotatedMethods) throws Exception {
-        final Object instance = handler.getDeclaredConstructor().newInstance();
-        for (Method method : annotatedMethods) {
-            putHandlerExecution(instance, method);
-        }
-    }
-
     private void putHandlerExecution(final Object instance, final Method method) {
         final RequestMapping requestMapping = method.getAnnotation(RequestMapping.class);
-        final String uri = requestMapping.value();
 
-        for (RequestMethod requestMethod : requestMapping.method()) {
-            final HandlerKey handlerKey = new HandlerKey(uri, requestMethod);
-            final HandlerExecution handlerExecution = new HandlerExecution(instance, method);
+        final List<HandlerKey> handlerKeys = getHandlerKeys(requestMapping);
+        final HandlerExecution handlerExecution = new HandlerExecution(instance, method);
 
+        for (HandlerKey handlerKey : handlerKeys) {
             handlerExecutions.put(handlerKey, handlerExecution);
         }
     }
+
+    private List<HandlerKey> getHandlerKeys(RequestMapping requestMapping) {
+        return Arrays.stream(requestMapping.method())
+            .map(it -> new HandlerKey(requestMapping.value(), it))
+            .collect(Collectors.toUnmodifiableList());
+    }
```

Comment:
```suggestion

    private List<HandlerKey> getHandlerKeys(final RequestMapping requestMapping) {
        return Arrays.stream(requestMapping.method())
            .map(it -> new HandlerKey(requestMapping.value(), it))
            .collect(Collectors.toUnmodifiableList());
    }
```
final 컨밴션이 자주 빠트리기 쉬운것같아요 😢

Reply:
계속 빼먹네요... 좀더 꼼꼼하게 봤어야 하는데 죄송합니다😢 
인텔리제이 내에서 final을 넣도록 자동화하는 방식이 있는 것으로 알고 있는데 만약 나중에 final을 붙이는 컨벤션을 계속 사용하게된다면 해당 기능을 사용하는 것이 좋을 것 같아요.